Dev/peripherial committee csv parser

wszostak opened the pull request.

wszostak committed:
Peripherial Committee csv parser

wszostak committed:
Created .gitignore file

wszostak committed:
Test allowedToVote mapping

wszostak committed:
More tests

 - catch exception when incomplete line given

 - check exception when file not found

 - Allow quotaion marks

Sitwa commented on java/openpkw-services/src/main/java/org/openpkw/mapping/PeripheralCommitteeFieldSetMapper.java:
```diff
@@ -0,0 +1,41 @@
+            peripheralCommittee.setType(fieldSet.readString(5));
+            peripheralCommittee.setName(fieldSet.readString(15));
+            try {
+                peripheralCommittee.setAllowedToVote(fieldSet.readLong(16));
```
Tu chyba lepszym pomysłem było by chyba zrobienie metody sprawdzającej z odpowiednią nazwą np. isAllowedToVote() i puszczenie ifa.  Łączenie logiki w try, catch jest słabe, powinno to służyć wyłączenie do obsługi wyjątków. Do zastanowienia też czy warunek powinien być w tym try- catch.

Sitwa commented on java/openpkw-services/src/main/java/org/openpkw/mapping/PeripheralCommitteeFieldSetMapper.java:
```diff
@@ -0,0 +1,41 @@
+                peripheralCommittee.setAllowedToVote(0);
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println(Arrays.toString(fieldSet.getValues()));
```
Musimy zrobić jakieś logowanie, puszczanie wyjątku na konsolkę jest kiepskie. Myślę, że temat na oddzielnego taska, przynajmniej wbijmy tu TODO i myślę, że ustawienie jakiegoś loggera jest priorytetem. Potem będzie dużo do zmiany jak będziemy z tym zwlekać.

wszostak replied:
Jak najbardziej się zgadzam.

Sitwa commented on java/openpkw-services/src/main/java/org/openpkw/services/CsvParser.java:
```diff
@@ -0,0 +1,37 @@
+        BufferedReader br = new BufferedReader(new FileReader(file));
+
+        int lineNumber = 0;
+        for(String line; (line = br.readLine()) != null; ) {
```
line = br.readLine()) != null;  
To powinno iść do nowej metody z odpowiednią nazwą. Poza tym myślę, że powinniśmy użyć tu jakiejś klasy do Stringów, chociażby StringUtils.isEmpty() z apacha. Sprawdza od razu czy String jest pusty i czy jest nullem.

wszostak replied:
To jest w pętli. Wyciągnięcie tego do jakiejś metody chyba nie poprawi czytelności. Sprawdzam `nulla` bo `readLine` to zwraca jeżeli nie ma już nic więcej do czytania. Teoretycznie pusty string jest poprawną wartością

Sitwa commented on java/openpkw-services/src/test/java/org/openpkw/services/PeripherialCommitteeCsvParserTest.java:
```diff
@@ -0,0 +1,67 @@
+        try {
+            result = csvParser.parse(tmpFile);
+        } catch (Exception e) {
+            e.printStackTrace();
```
ta sama uwaga względem logowania

wszostak replied:
parse rzuca `Exception` bo `mapLine` z `package org.springframework.batch.item.file.LineMapper` rzuca `Exception`...

wszostak replied:
To jest test, więc plik powinien być w resourcach, ale wywaliłem `try/catch` bo też mi się nie podobało.

Sitwa commented on java/openpkw-services/src/test/java/org/openpkw/services/PeripherialCommitteeCsvParserTest.java:
```diff
@@ -0,0 +1,67 @@
+            e.printStackTrace();
+        }
+
+        Assert.assertEquals(result.get(0).getPkwId(), "020101");
```
result.get(0) możemy wyciągnąć do jakiejś zmiennej.Test będzie czytelniejszy.

Sitwa commented on java/openpkw-services/src/test/java/org/openpkw/services/PeripherialCommitteeCsvParserTest.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenFileNotFound() {
```
@Test(expected=FileNotFoundException.class) 

Coś takiego powinno skrócić test i zrobić go bardziej czytelnym. Poza tym, nie będziemy rzucać stacktrace na konsolę podczas wykonywania testu o ile dobrze pamiętam.

wszostak replied:
Szczerze powiedziawszy, próbowałem tak ale nie chciało mi się skompilować. Spróbuję raz jeszcze.

wszostak replied:
Sprawdziłem, poprawiłem...

wszostak committed:
CR fixes

 - Added logger
 - Removed extranous exception catching
 - Mapping for next fields
 - Skipping failed lines
 - Tests extended

wszostak committed:
Removed exception catching from test

wszostak committed:
Test beauify

wszostak committed:
Use proper types

wszostak committed:
Change logged message format

wszostak committed:
Solved Unchecked cast warning

By the way enabled lint while compilation

wszostak committed:
Removed dependencies from openpkw-core

wszostak committed:
Tune PeripherialCommitteeParser to new CSV files
